pkg/window/strategy/fixed/fixed.go

```diff
@@ -147,12 +148,13 @@ func (w *Windower) InsertWindow(tw window.TimedWindow) {
 // returns a list of TimedWindowRequests, each request contains the window operation and the window
 // which needs to be closed.
 func (w *Windower) CloseWindows(time time.Time) []*window.TimedWindowRequest {
+	// TODO - we are updating both active and closed windows. do we need lock? same question for other windower methods.
```

Comment:
```suggestion
	// FIXME - we are updating both active and closed windows. do we need lock? same question for other windower methods.
```

data races are severe.